frb_codegen/src/ir/field.rs

```diff
@@ -18,15 +20,32 @@ impl IrField {
             self.ty.dart_required_modifier()
         }
     }
-    pub fn field_default(&self, freezed: bool) -> String {
+    pub fn field_default(&self, freezed: bool, config: Option<&Opts>) -> String {
         self.default
             .as_ref()
             .map(|r#default| {
                 let r#default = match r#default {
                     DefaultValues::Str(lit)
                         if !matches!(&self.ty, IrType::Delegate(IrTypeDelegate::String)) =>
                     {
-                        lit.value().into()
+                        // Convert the default value to Dart style.
+                        if config.is_some() && config.unwrap().dart_enums_style {
+                            let value = lit.value();
+                            let mut split = value.split('.');
+                            let enum_name = split.next().unwrap();
+
+                            let variant_name = split.next().unwrap().to_string();
+                            let variant_name =
+                                if crate::utils::check_for_keywords(&[variant_name.to_case(Case::Camel)]).is_err() {
+                                    variant_name.to_case(Case::Pascal)
+                                } else {
+                                    variant_name.to_case(Case::Camel)
+                                };
```

Comment:
uber-nit: If we use the approach above, here we can reuse that avoid-collision function